Libraries/LibWeb/Layout/Node.cpp

```diff
@@ -1053,6 +1053,12 @@ GC::Ref<NodeWithStyle> NodeWithStyle::create_anonymous_wrapper() const
     wrapper->mutable_computed_values().set_text_decoration_thickness(computed_values().text_decoration_thickness());
     wrapper->mutable_computed_values().set_text_decoration_color(computed_values().text_decoration_color());
     wrapper->mutable_computed_values().set_text_decoration_style(computed_values().text_decoration_style());
+
+    // CSS 2.2 9.2.1.1 creates anonymous block boxes, but 9.4.1 states inline-block creates a BFC.
+    // Set wrapper to inline-block to participate correctly in the IFC within the parent inline-block.
+    if (display().is_inline_block() && !last_child()) {
```

Comment:
Instead of `!last_child()`, `!has_children()` might be more readable version of the same condition.